### PR TITLE
Seed the data SN-385

### PR DESCRIPTION
**Pull request by vasiliy-chefonov**
https://saritasa.atlassian.net/browse/SN-385

### CODE REVIEW

File: demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        if (Count < 1)
+        {
+            logger.LogInformation("Nothing to generate.");
+            return;
+        }
+
+        // Run default seeder if no specific.
+        var classType = !string.IsNullOrEmpty(Name) ?
+            Type.GetType($"{SeederNamespace}.{Name}Seeder", false, true) :
+            typeof(AddressesSeeder);
```
Comment (AlexanderMakarov124) [inline demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs:68]: I think this version is more readable
```suggestion
        var classType = !string.IsNullOrEmpty(Name) 
            ? Type.GetType($"{SeederNamespace}.{Name}Seeder", false, true) 
            : typeof(AddressesSeeder);
```

File: demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        return ctor.Invoke(inputParamObjects.ToArray());
+    }
+
+    /// <summary>
+    /// Parse arguments like "--argName=10" to dictionary "argname":10.
+    /// </summary>
+    private IDictionary<string, string> ParseRemainingArguments()
+    {
+        var dict = new Dictionary<string, string>();
+        if (RemainingArguments == null) return dict;
```
Comment (AlexanderMakarov124) [inline demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs:123]: By code style we have to add braces here
image

File: demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        {
+            var args = remainingArgument.TrimStart('-').Split(new[] { '=', ':' }, 
+                StringSplitOptions.RemoveEmptyEntries);
+            
+            if (args.Length != 2)
+            {
+                continue;
+            }
+            
+            dict[args[0].ToLower()] = args[1];
```
Comment (AlexanderMakarov124) [inline demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs:135]: A lot of magic numbers here, maybe add constants instead?

File: demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            if (!string.IsNullOrEmpty(nameKey) && args.TryGetValue(nameKey, out var arg))
+            {
+                var tc = TypeDescriptor.GetConverter(parameterInfo.ParameterType);
+                value = tc.ConvertFrom(arg);
+            }
+            inputParams.Add(value);
+        }
+
+        // Calling.
+        var retObject = seedMethod.Invoke(obj, inputParams.ToArray());
```
Comment (AlexanderMakarov124) [inline demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs:176]: I think `result` will be more readable then `retObject`. What do you think?

File: demo/Saritasa.NetForge.Demo.Net7/Models/Product.cs [hunk 5]
```diff
@@ -36,22 +36,22 @@ public class Product
     /// <summary>
     /// The weight of the product in grams.
     /// </summary>
-    public double WeightInGrams { get; set; }
+    public float WeightInGrams { get; set; }
```
Comment (ko-vasilev) [inline demo/Saritasa.NetForge.Demo.Net7/Models/Product.cs:39]: Why did you change it to float from double?

**Review by AlexanderMakarov124 — APPROVED** [review]

File: demo/Saritasa.NetForge.Demo.Net7/Startup.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        services.AddHealthChecks().AddNpgSql(connectionString);
+        
+        // Identity.
+        services.AddIdentity<User, IdentityRole>()
+            .AddEntityFrameworkStores<ShopDbContext>()
+            .AddDefaultTokenProviders();
+        services.Configure<IdentityOptions>(new IdentityOptionsSetup().Setup);
+        
+        // Register NetForge.
+        services.AddNetForge(optionsBuilder =>
```
Comment (ko-vasilev) [inline demo/Saritasa.NetForge.Demo.Net7/Startup.cs:52]: Maybe move this to a separate method?

**Review by ko-vasilev — APPROVED** [review]

File: demo/Saritasa.NetForge.Demo.Net7/Models/Product.cs (hunk 5 above)
Comment (vasiliy-chefonov) [reply demo/Saritasa.NetForge.Demo.Net7/Models/Product.cs:39]: I think we just don't need such precision. Float is enough for us.

File: demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs (hunk 3 above)
Comment (vasiliy-chefonov) [reply demo/Saritasa.NetForge.Demo.Net7/Commands/Seed.cs:135]: I suggest just assigning them to variables.
